### ISSUE TITLE
Aftergraces synced with other unrelated notes will not receive listeners

### ISSUE THREAD

**Issue by NyuBlara** [scrime-u-bordeaux/web-midifile-performer#100]
As an example : K276-1, measure 40. 

![image](https://github.com/user-attachments/assets/6ade5fb9-9cb0-49ec-994d-3d6d95538423)

The first of these two aftergraces is triggered in this set, along with the two left-hand eighth notes. It is not found when setting up noteheads, and thus receives no listener. 

In general, aftergraces should not be synced with anything. So the best way to fix this would actually be to force a small delay between them and any other events when parsing the XML file.

**Comment by josephlarralde** [comment]
Yep, thinking a little bit overhead I'm seriously considering rewriting the C++ library in javascript to extend it more easily since we don't have other active c++ bindings ... also I recently experimented behaviours that smelled like memory leaks as discussed in a previous issue of the library. We could then implemend setTimeout and setInterval based logic for trills etc directly in the js library

**Comment by NyuBlara** [comment]
IMHO, this is above the library level. Grace notes only exist in the MusicXML realm ; they are only a notation artifact, which has been mostly abandoned in modern music, and to a MIDI file, they are ordinary short notes. The same is true of trills and other such concepts : they are not recognizable in a MIDI file, and thus cannot be dealt with at that level of abstraction.

**Comment by josephlarralde** [comment]
Yes, but : with it we are currently limited to action -> reaction (1 gesture produces 1 synchronous event and that's all)

**Comment by NyuBlara** [comment]
In that sense, yes, we could reflect on this ; but I feel it's unrelated to the present issue.

**Comment by josephlarralde** [comment]
Yeah, sorry I'm a serial thread hacker ...
